ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/LedgerDao.scala

```diff
@@ -242,9 +271,42 @@ trait LedgerWriteDao extends AutoCloseable with ReportsHealth {
       externalOffset: Option[ExternalOffset]
   ): Future[Map[PersistenceResponse, Int]]
 
+  /**
+    * Store a package upload entry confirmation or rejection
+    *
+    * @param offset
+    * @param newLedgerEnd
+    * @param externalOffset
+    * @param entry
+    * @return
+    */
+  def storePackageUploadEntry(
```

Comment:
Should be combined with `uploadLfPackages`.

Reply:
right now uploadLfPackages is called once per archive (in the case of a submission with multiple DARs) and then storePackageUploadEntry is called once at the end.  Or in the case of rejected packages uploadLfPackages is never called and storePackageUploadEntry is called once for the reject.

So similar to the above about atomicity can we track a change here as a follow-on PR?

Reply:
i take it back this one is doable - we basically should do the package upload and the insert of the entries within the one `uploadLfPackages` method

Reply:
i did most of the changes for this locally but what we don't have in teh PublicPackageUploaded event is the info on whether the package is accepted or not
![image](https://user-images.githubusercontent.com/25179017/70110356-b59bbf80-161c-11ea-8397-b64035719267.png)